pkg/autoscaler/statforwarder/processor.go

```diff
@@ -39,6 +47,25 @@ type bucketProcessor struct {
 	accept statProcessor
 }
 
+func newForwardProcessor(logger *zap.SugaredLogger, bkt, holder, podDNS, svcDNS string) *bucketProcessor {
+	// First try to connect via Pod IP address synchronously. If the connection can
+	// not be established within `establishTimeout`, we assume the pods can not be
+	// accessed by IP address. Then try to connect via Pod IP address asynchronously
+	// to avoid nil check.
+	logger.Infof("Connecting to Autoscaler bucket at ", podDNS)
+	c, err := newConnection(podDNS, logger)
+	if err != nil {
+		logger.Info("Autoscaler pods can't be accessed by IP address. Connecting to Autoscaler bucket at ", svcDNS)
+		c = websocket.NewDurableSendingConnection(svcDNS, logger)
```

Comment:
why not wait here as well?

Reply:
If we wait, the returned connection could be nil. Then in `process` function, we have to check whether it's nil nor not. If nil, create the connection and wait again. I'd prefer to letting the connection keep trying connecting for SVC if there's error instead of waiting, closing, recreating.

Reply:
Do we assume that if there's no network issue, the connection should be connected within 0.5 second?

Reply:
I think it's reasonable assumption. If another autoscaler declared itself ready and kubelet saw it and propagated to the endpoints, I think we can assume we should be able to connect to it. And 500ms is a lot for intracluster connection.

Reply:
done